packages/state_queue/lib/src/state_queue.dart

```diff
@@ -176,14 +185,19 @@ abstract class StateQueue<T> extends ValueNotifier<T>
   /// of the bloc can not be awaited properly.
   @protected
   void run(StateUpdater<T> updater) {
-    _taskQueue.sink.add(_UpdaterEntry(updater));
+    final entry = _UpdaterEntry(updater);
```

Comment:
How about passing `done` to the `_UpdaterEntry` so we don't create the extra completer?

Reply:
Done